azure_functions_worker/dispatcher.py

```diff
@@ -294,16 +292,54 @@ async def _handle__worker_init_request(self, req):
                 result=protos.StatusResult(
                     status=protos.StatusResult.Success)))
 
-    async def _handle__worker_status_request(self, req):
+    async def _handle__worker_status_request(self, request):
         # Logging is not necessary in this request since the response is used
         # for host to judge scale decisions of out-of-proc languages.
         # Having log here will reduce the responsiveness of the worker.
         return protos.StreamingMessage(
-            request_id=req.request_id,
+            request_id=request.request_id,
             worker_status_response=protos.WorkerStatusResponse())
 
-    async def _handle__function_load_request(self, req):
-        func_request = req.function_load_request
+    async def _handle__functions_metadata_request(self, request):
+        logger.info(f"Received FunctionMetadataRequest: {self.request_id}")
+
+        metadata_request = request.functions_metadata_request
+        directory = metadata_request.function_app_directory
+        function_path = os.path.join(directory, SCRIPT_FILE_NAME)
+
+        if not os.path.exists(function_path):
+            # Fallback to legacy model
+            return protos.StreamingMessage(
+                request_id=request.request_id,
+                function_metadata_response=protos.FunctionMetadataResponse(
+                    use_default_metadata_indexing=True,
+                    result=protos.StatusResult(
+                        status=protos.StatusResult.Success)))
+
+        try:
+            fx_metadata_results = []
+            indexed_functions = loader.index_function_app(function_path)
+            if indexed_functions:
+                fx_metadata_results = self._process_indexed_function(
+                    indexed_functions)
+
+            return protos.StreamingMessage(
+                request_id=request.request_id,
+                function_metadata_response=protos.FunctionMetadataResponse(
+                    function_metadata_results=fx_metadata_results,
+                    result=protos.StatusResult(
+                        status=protos.StatusResult.Success)))
+
+        except Exception as ex:
```

Comment:
Any specific cases where we could expect the exception? If yes, then might as well narrow down the exceptions as well, for better logging.